src/controllers/controller.cpp

```diff
@@ -143,4 +147,6 @@ void Controller::receive(const QByteArray data, mixxx::Duration timestamp) {
         QJSValue incomingDataFunction = m_pEngine->wrapFunctionCode(function, 2);
         m_pEngine->executeFunction(incomingDataFunction, data);
     }
+
+    m_pEngine->handleInput(data, timestamp);
```

Comment:
Here is the place where the dispatcher should directly call the js callbacks. 
For standard button connections we even bypass the js round trip by altering COs directly. 
This will also help to clutter the js mapping. This need to provide callbacks for the difficult things only.  

The mapping can still optional provide an "incomingData" function to perform difficult tasks.

Reply:
No, the dispatcher is JS. There are no plans to change this.